extension/src/experiments/webview/messages.ts

```diff
@@ -265,6 +267,19 @@ export class WebviewMessages {
     )
   }
 
+  private async redirectToSetup() {
+    sendTelemetryEvent(
+      EventName.VIEWS_EXPERIMENTS_TABLE_REDIRECT_TO_SETUP,
+      undefined,
+      undefined
+    )
+
+    await commands.executeCommand(RegisteredCommands.SETUP_SHOW_EXPERIMENTS)
+
+    const webview = this.getWebview()
+    webview?.dispose()
```

Comment:
Is there a better way to handle a "redirect"?

Reply:
Why send the message to the webview to send a message back? could add code in `sendWebviewMessage` to close the webview/open setup if no data is detected. This would mean the logic in the webview could also be simplified as hasData would never be false (could remove it)

Reply:
less code to maintain, less to test, etc

Reply:
Good idea!

Reply:
Done!